commands/account/auth.ts

```diff
@@ -0,0 +1,183 @@
+import { Argv, ArgumentsCamelCase } from 'yargs';
+import {
+  loadConfig,
+  getConfigPath,
+  configFileExists,
+  updateAccountConfig,
+  writeConfig,
+  createEmptyConfigFile,
+  deleteEmptyConfigFile,
+  getConfigDefaultAccount,
+} from '@hubspot/local-dev-lib/config';
+import { logger } from '@hubspot/local-dev-lib/logger';
+import {
+  getAccessToken,
+  updateConfigWithAccessToken,
+} from '@hubspot/local-dev-lib/personalAccessKey';
+import { ENVIRONMENTS } from '@hubspot/local-dev-lib/constants/environments';
+import { toKebabCase } from '@hubspot/local-dev-lib/text';
+import { Environment } from '@hubspot/local-dev-lib/types/Config';
+import { CLIAccount_NEW } from '@hubspot/local-dev-lib/types/Accounts';
+
+import { addConfigOptions, addGlobalOptions } from '../../lib/commonOpts';
+import { handleExit } from '../../lib/process';
+import { logError } from '../../lib/errorHandlers/index';
+import { i18n } from '../../lib/lang';
+import { trackCommandUsage } from '../../lib/usageTracking';
+import { addTestingOptions } from '../../lib/commonOpts';
+import { personalAccessKeyPrompt } from '../../lib/prompts/personalAccessKeyPrompt';
+import { cliAccountNamePrompt } from '../../lib/prompts/accountNamePrompt';
+import { setAsDefaultAccountPrompt } from '../../lib/prompts/setAsDefaultAccountPrompt';
+import { EXIT_CODES } from '../../lib/enums/exitCodes';
+import { uiFeatureHighlight } from '../../lib/ui';
+import { CommonArgs, ConfigArgs } from '../../types/Yargs';
+
+const i18nKey = 'commands.account.subcommands.auth';
+
+async function createOrUpdateConfig(
+  env: Environment,
+  doesConfigExist: boolean,
+  account?: number
+): Promise<CLIAccount_NEW | null> {
+  try {
+    const { personalAccessKey } = await personalAccessKeyPrompt({
+      env,
+      account,
+    });
+    const token = await getAccessToken(personalAccessKey, env);
+    const defaultName = token.hubName ? toKebabCase(token.hubName) : null;
+
+    const name = doesConfigExist
+      ? undefined
+      : (await cliAccountNamePrompt(defaultName)).name;
+
+    const updatedConfig = await updateConfigWithAccessToken(
+      token,
+      personalAccessKey,
+      env,
+      name,
+      !doesConfigExist
+    );
+
+    if (!updatedConfig) return null;
+
+    if (doesConfigExist && !updatedConfig.name) {
+      updatedConfig.name = (await cliAccountNamePrompt(defaultName)).name;
+    }
+
+    if (doesConfigExist) {
+      updateAccountConfig({
+        ...updatedConfig,
+        environment: updatedConfig.env,
+        tokenInfo: updatedConfig.auth!.tokenInfo,
+      });
+      writeConfig();
+    }
+
+    return updatedConfig as CLIAccount_NEW;
```

Comment:
I'm casting `updatedConfig` as `CLIAccount_NEW` here, because otherwise, I'd receive a TypeScript error that `accountId` doesn't exist on type `CLIAccount`. This is because `CLIAccount` can be either `CLIAccount_NEW` or `CLIAccount_DEPRECATED`, and `accountId` doesn't exist on `CLIAccount_DEPRECATED`. 

Is this an acceptable solution, or am I missing anything? I'm assuming we don't want to make any major changes to types for the deprecated config.

Reply:
iirc we made a util to handle this situation. It's `getAccountIdentifier`, which can be used to safely read the id from an account regardless of the config type.

Reply:
I'm a little unclear about how I would use `getAccountIdentifier` in this case, since we're returning all the account info and not just the ID. I don't think reassigning `updatedConfig.accountId` using `getAccountIdentifier` would help, correct? Because the issue is the property name.

Reply:
Oh, I was assuming the error was coming from [here](https://github.com/HubSpot/hubspot-cli/pull/1365/files#diff-f0f185d80a68501d80d813e663386e095d242d8b6bc4836d250897471ef14735R139) where you're accessing `accountId` from the config object.

Is the error coming from the return type of this function instead? If so, could it be fixed by switching the return type to `CLIAccount` instead of `CLIAccount_NEW`?

Reply:
Addressed in https://github.com/HubSpot/hubspot-cli/pull/1365/commits/c7dcc882173084d4594bea4cb6294fc4ab993d14